docs/src/architecture/09_architecture_decisions/0007-api-versions.md

```diff
@@ -0,0 +1,97 @@
+---
+    title: 0007 Api Versions
+    adr:
+        author: Steven Johnson
+        created: 16-Oct-2024
+        status:  proposed
+    tags:
+        - api
+---
+
+## Context
+
+The Catalyst Voices backend service, known as Catalyst Gateway, provides an HTTP API to front end services.
+
+It is required that the API be:
+
+* Stable enough for Frontend integrations to be reliable;
+* Flexible enough to allow endpoints to evolve, be modified or removed over time.
+
+A secondary consideration is that wherever practical, the API be defined such that it can be replicated on the Hermes Engine.
+
+## Assumptions
+
+That Catalyst Gateway will provide its API's with an HTTP API, defined by an OpenAPI specification.
+
+## Decision
+
+All Individual API Endpoints will be named and versioned according to the following rules:
+
+1. All endpoints URIs will be under the `/api` path in the backend.
+2. The general structure of any endpoint will be `/api/<version>/<path>`
+3. `<version>` defines a version of the endpoint supplied by a `<path>` and is **NOT** aligned with any other versions.
+4. `<version>` **MUST** either be `draft` or matching the regular expression `^v(?!0)\d{1,}$`[^1]
+5. `<path>` Can be any path of at least one element as required for the endpoint.
+6. **ALL** new endpoints start as `draft` they are promoted to a versioned endpoint by a deliberate decision.
+7. New endpoints **MUST NOT** be versioned other than `draft` in their initial PR.
```

Comment:
Can we please consider adding an example of such a `draft` endpoint as we did above `/api/<version>/<path>`?

Reply:
Examples added